go/vt/topo/etcd2topo/watch.go

```diff
@@ -54,23 +54,27 @@ func (s *Server) Watch(ctx context.Context, filePath string) (*topo.WatchData, <
 		Version:  EtcdVersion(initial.Kvs[0].ModRevision),
 	}
 
+	// Create an outer context that will be canceled on return and will cancel all inner watches.
+	outerCtx, outerCancel := context.WithCancel(ctx)
+
 	// Create a context, will be used to cancel the watch on retry.
-	watchCtx, watchCancel := context.WithCancel(ctx)
+	watchCtx, watchCancel := context.WithCancel(outerCtx)
 
 	// Create the Watcher.  We start watching from the response we
 	// got, not from the file original version, as the server may
 	// not have that much history.
 	watcher := s.cli.Watch(watchCtx, nodePath, clientv3.WithRev(initial.Header.Revision))
 	if watcher == nil {
 		watchCancel()
+		outerCancel()
 		return nil, nil, vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "Watch failed")
 	}
 
 	// Create the notifications channel, send updates to it.
 	notifications := make(chan *topo.WatchData, 10)
 	go func() {
 		defer close(notifications)
-		defer watchCancel()
```

Comment:
Curious why we removed this line but call watchCancel() before outerCancel() in other places like like 192?